### ISSUE TITLE
Error when using with other webparts that use the package @micorsoft/mgt-react or @micorsoft/mgt-spfx library

### ISSUE THREAD

**Issue by Frau87** [microsoft-search/pnp-modern-search#2795]
**Version used**
Ex: 4.8

**Describe the bug**
I developed a webpart that uses Person mgt-react component. It use Porviders from @micorsoft/mgt-spfx for athentication. Once installed in SC with some pnp search webpart installed, start recieving a lot of error on both webparts (mine and the search)

like the one described here [https://github.com/microsoftgraph/microsoft-graph-toolkit/issues/229](https://github.com/microsoftgraph/microsoft-graph-toolkit/issues/229) and other related to the Global Provider object. I think it might be related to the different implememtation of Providers in @micorsoft/mgt shifting to @micorsoft/mgt-spfx. But using @micorsoft/mgt-react i'm not been able to let it work with @micorsoft/mgt standard package.

Resolved fixing on pnp search all reference to @micorsoft/mgt shifting to @micorsoft/mgt-spfx.

Think this error might occur with other custom webparts too. Suggest to migrate also pnp-search to the spfx package

(The issue occurs only when using MSGraph Toolkit on Pnp Search of course)

Reported here for anyone with the same issue. Hope it would help.

**Comment by jimmywim** [comment]
Is there any movement on correcting the mgt dependencies in this solution?

**Comment by jimmywim** [comment]
@wobba It's as described by @Frau87 at the top - a separate web part using MGT with the SharePoint provider, as soon as you turn on MGT in Search Results the web part produces the error:

```
ERROR:
Failed to execute 'define' on 'CustomElementRegistry': the name "mgt-mock-provider" has already been used with this registry
```

As it's referencing the mock provider, then it sounds like your fix will sort this. Just gotta wait for 4.9 now 😊

**Comment by Frau87** [comment]
I'd fix it like this

![image](https://user-images.githubusercontent.com/20320305/236498292-9a87ad46-2c5c-4b13-adb0-f19274e9a156.png)

![image](https://user-images.githubusercontent.com/20320305/236498366-8c3ee7da-aee1-4827-b9a3-0cd4feeb4a8e.png)

Of course you have to install the package via NPM and also install the mgt-spfx shared library at Tenant/site level

**Comment by wobba** [comment]
@Frau87 if that works we can change. It used to break with webpack - which is why we didn't change to the spfx import. @Frau87 did you try to build the dev branch with my fix?

**Comment by Frau87** [comment]
Nope. I'll give It a try ASAP

**Comment by Frau87** [comment]
@wobba tried your fix from develoeper branch. Seems to work!

**Comment by Frau87** [comment]
@wobba the fix in your dev branch seems to work!

**Comment by wobba** [comment]
Then I'll close this issue as being resolved in th v4.9 when we build it.
